src/DynamicData/Cache/ObservableCache.cs

```diff
@@ -240,4 +322,103 @@ private void InvokePreview(ChangeSet<TObject, TKey> changes)
             }
         }
     }
+
+    private void InvokeCountNext()
+    {
+        lock (_locker)
+        {
+            if (_countChanged.IsValueCreated)
+            {
+                _countChanged.Value.OnNext(_readerWriter.Count);
+            }
+        }
+    }
+
+    private void ResumeCount()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Count without Suspend Count instance");
+            _suspensionTracker.Value.ResumeCount();
+        }
+    }
+
+    private void ResumeNotifications()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Notifications without Suspend Count instance");
+            _suspensionTracker.Value.ResumeNotifications();
+        }
+    }
+
+    private class SuspensionTracker(Action<ChangeSet<TObject, TKey>> onResumeNotifications, Action onResumeCount) : IDisposable
+    {
+        private readonly BehaviorSubject<bool> _notifySuspendSubject = new(false);
+
+        private readonly Action<ChangeSet<TObject, TKey>> _onResumeNotifications = onResumeNotifications;
+
+        private readonly Action _onResumeCount = onResumeCount;
+
+        private IEnumerable<Change<TObject, TKey>> _pending = Enumerable.Empty<Change<TObject, TKey>>();
+
+        private int _countSuspendCount;
+
+        private int _notifySuspendCount;
+
+        public bool CountSuspended => _countSuspendCount > 0;
+
+        public bool NotificationsSuspended => _notifySuspendCount > 0;
+
+        public IObservable<bool> NotificationsSuspendedStatus => _notifySuspendSubject;
+
+        public void AddPending(IEnumerable<Change<TObject, TKey>> changes)
```

Comment:
VERY minor nitpick, but I would suggest `AddPendingChanges` or `EnqueueChanges` or something. Just `AddPending` immediately begs the question "pending what?" which I know is obvious when looking at the method signature, but not necessarily so much when looking at calling-code.

Reply:
EnqueueChanges is better.